walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> availableTimesForMandatoryAttendeesOnly =
+      getAvailableTimes(events, request, request.getAttendees());
+    ArrayList<TimeRange> availableTimesForOptionalAttendeesOnly = 
+      getAvailableTimes(events, request, request.getOptionalAttendees());
+    
+    ArrayList<TimeRange> availableTimesForAllAttendees = 
+      new ArrayList<TimeRange>(availableTimesForMandatoryAttendeesOnly);
+    availableTimesForAllAttendees.retainAll(availableTimesForOptionalAttendeesOnly);
```

Comment:
For the following case:

```
availableTimesForMandatoryAttendeesOnly: [WHOLE_DAY],
availableTimesForOptionalAttendeesOnly: [[9:00, 10:00]],
```

Wouldn't the output be empty? Why not call ```getAvailableTimes``` again with just all the participants?